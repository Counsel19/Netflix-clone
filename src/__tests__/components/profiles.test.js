/* eslint-disable testing-library/no-node-access */
import React from "react";
import { render, screen } from "@testing-library/react";
import { Profiles } from "../../components";

describe("<Profiles />", () => {
  it("renders the <Profile /> with populated data", () => {
    const { container } = render(
      <Profiles>
        <Profiles.Title>Who is Watching</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture src='/images/users/1.jpg' data-testid="profile-picture"/>
            <Profiles.Name>Counsel</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(screen.getByText('Who is Watching')).toBeTruthy();
    expect(screen.getByTestId('profile-picture')).toBeTruthy();
    expect(screen.getByText('Counsel')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
    
  });

  it("renders the <Profiles /> with populated data but with a misc image", () => {
    const { container } = render(
        <Profiles>
          <Profiles.Title>Who is Watching</Profiles.Title>
          <Profiles.List>
            <Profiles.User onClick={() => {}}>
              <Profiles.Picture data-testid="profile-picture-misc"/>
              <Profiles.Name>Counsel</Profiles.Name>
            </Profiles.User>
          </Profiles.List>
        </Profiles>
      );
  
      expect(screen.getByText('Who is Watching')).toBeTruthy();
      expect(screen.getByTestId('profile-picture-misc')).toBeTruthy();
      expect(screen.getByText('Counsel')).toBeTruthy();
      expect(container.firstChild).toMatchSnapshot();
  });
});
