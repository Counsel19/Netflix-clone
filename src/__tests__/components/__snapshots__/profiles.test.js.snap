// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profile /> with populated data 1`] = `
<div
  class="profiles__Container-sc-1wt6s93-0 hAVqKu"
>
  <h1
    class="profiles__Title-sc-1wt6s93-1 csBSZM"
  >
    Who is Watching
  </h1>
  <ul
    class="profiles__List-sc-1wt6s93-2 kQuncN"
  >
    <li
      class="profiles__Item-sc-1wt6s93-5 bPaofq"
    >
      <img
        class="profiles__Picture-sc-1wt6s93-4 jDWBIK"
        data-testid="profile-picture"
        src="/images/users//images/users/1.jpg.png"
      />
      <p
        class="profiles__Name-sc-1wt6s93-3 cxsnpd"
      >
        Counsel
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders the <Profiles /> with populated data but with a misc image 1`] = `
<div
  class="profiles__Container-sc-1wt6s93-0 hAVqKu"
>
  <h1
    class="profiles__Title-sc-1wt6s93-1 csBSZM"
  >
    Who is Watching
  </h1>
  <ul
    class="profiles__List-sc-1wt6s93-2 kQuncN"
  >
    <li
      class="profiles__Item-sc-1wt6s93-5 bPaofq"
    >
      <img
        class="profiles__Picture-sc-1wt6s93-4 jDWBIK"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-sc-1wt6s93-3 cxsnpd"
      >
        Counsel
      </p>
    </li>
  </ul>
</div>
`;
